game/src/game/services/GameStateService.cpp

```diff
@@ -1195,73 +1260,113 @@ std::unordered_set<std::string> GameStateService::GetPlayerStaticNames()
-    Checkpoints checkpoints;
-    auto locations = checkpoints.GetCheckpoints();
+std::vector<std::pair<glm::vec3, std::string>>
+GameStateService::ReadCheckpointsFromJsonFile()
+{
+    std::vector<std::pair<glm::vec3, std::string>> data;
 
-    // locations.second = orientation, we do not care about that, ignore that
-    // for now
+    std::ifstream file("resources/powerup/PowerupSpawnPoint.jsonc");
+    if (!file.is_open())
+    {
+        debug::LogError("Powerup Spawn not happening");
+    }
 
-    for (const auto& l : locations)
+    std::string jsonStr((std::istreambuf_iterator<char>(file)),
+                        std::istreambuf_iterator<char>());
+    rapidjson::Document document;
+    document.Parse(jsonStr.c_str());
+
+    if (document.IsArray())
     {
-        // adding a random statement which will select to spawn ammo type or
-        // powerup at the lcoation randomly generating what kind of powerup
-        uint8_t selector = rand() % 2 + 1;
-        if (selector == 1)
+        for (rapidjson::SizeType i = 0; i < document.Size(); i++)
         {
-            // means ammo should spawn
-            uint8_t random_ammo = rand() % 5 + 1;
-            switch (random_ammo)
+            const rapidjson::Value& object = document[i];
+
+            if (object.HasMember("Location") && object.HasMember("Pickup_Type"))
             {
-                case 1:
-                    ammo_info_.push_back({AmmoPickupType::kBuckshot, l.first});
-                    break;
-                case 2:
-                    ammo_info_.push_back(
-                        {AmmoPickupType::kDoubleDamage, l.first});
-                    break;
-                case 3:
-                    ammo_info_.push_back(
-                        {AmmoPickupType::kExploadingBullet, l.first});
-                    break;
-                case 4:
-                    ammo_info_.push_back(
-                        {AmmoPickupType::kIncreaseFireRate, l.first});
-                    break;
-                case 5:
-                    ammo_info_.push_back(
-                        {AmmoPickupType::kVampireBullet, l.first});
-                    break;
+                const rapidjson::Value& location = object["Location"];
+                const rapidjson::Value& pickup_type = object["Pickup_Type"];
+
+                if (location.IsString() && pickup_type.IsString())
+                {
+                    const char* location_str = location.GetString();
+                    const char* pickup_type_str = pickup_type.GetString();
+
+                    std::stringstream ss(location_str);
+                    float x, y, z;
+                    ss >> x >> y >> z;
+
+                    if (!ss.fail())
+                    {
+                        glm::vec3 location(x, y, z);
+                        data.push_back(std::make_pair<>(
+                            location, pickup_type.GetString()));
+                    }
```

Comment:
Rather than parsing the string manually, to get the components out, try an array of floats. In my PR I also made some utility functions for parsing a Vec3 from JSON cause I needed it https://github.com/raduschirliu/cpsc585/pull/175/files#diff-8f8eb6da230b43b0369cbbcd32a2c1769abf23ce23994d0d09ddad01656368e4R37